plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/PredicatePushdownController.java

```diff
@@ -62,6 +66,18 @@ public interface PredicatePushdownController
         return new DomainPushdownResult(simplifiedDomain, domain);
     };
 
+    static PredicatePushdownController pushdownDiscreteValues(Type type)
+    {
+        return (session, domain) -> {
+            Optional<Collection<Object>> expandedRange = domain.getValues().tryExpandRanges(getDomainCompactionThreshold(session));
+            if (expandedRange.isPresent()) {
+                Domain convertedDiscreteDomain = Domain.multipleValues(type, expandedRange.get().stream().toList());
+                return new PredicatePushdownController.DomainPushdownResult(convertedDiscreteDomain, domain);
```

Comment:
`DomainPushdownResult(convertedDiscreteDomain, Domain.all(domain.getType()))`

Reply:
thank you! updated